hl7_transformer/hl7_transformer/application.py

```diff
@@ -74,18 +74,41 @@ def _process_message(
         msh_segment = hl7_msg.msh
         logger.debug(f"Message ID: {msh_segment.msh_10.value}")
 
+        transformations_applied = []
+
         created_datetime = msh_segment.msh_7.value
 
         transformed_datetime = transform_datetime(created_datetime)
         msh_segment.msh_7.value = transformed_datetime
 
+        transformations_applied.append(
+                f"DateTime transformed from {created_datetime} to {transformed_datetime}"
+            )
+
+        if hl7_msg.pid:
+            pid_segment = hl7_msg.pid
```

Comment:
in the other transformer we work on the assumption that `pid` might not even be there at all - in which case you'll get an `AttributeError`. If this is a possibility it would be better to do `getattr(hl7_msg, "pid", None)`
```suggestion
       pid_segment = getattr(hl7_msg, "pid", None)
       if pid_segment is None:
          return
       
```

Reply:
also could there be multiple PID segments for PHW?

Reply:
I haven't heard about HL7 messages with multiple PIDs, but maybe do you know if those can exist?